src/hal/components/toggle2nist.comp

```diff
@@ -4,61 +4,68 @@ description
 """
 Toggle2nist can be used with a momentary push button 
 to control a device that has separate on and off inputs
-and an is-on output. 
+and an is-on output.
+A debounce delay in cycles can be set. (default = 2)
 
 \[bu] On a rising edge on pin \\fIin\\fR when \\fIis-on\\fR is low: It sets \\fIon\\fR until \\fIis-on\\fR becomes high.
 
 \[bu] On a rising edge on pin \\fIin\\fR when \\fIis-on\\fR is high: It sets \\fIoff\\fR until \\fIis-on\\fR becomes low.
 
 
-       ┐     ┌─────xxxxxxxxxxxx┐           ┌─────xxxxxxxxxxxx┐
+       ┐     ┌─────xxxxxxxxxxxx┐           ┌─────xxxxxxxxxxxx┐
 .br
-in   : └─────┘     xxxxxxxxxxxx└───────────┘     xxxxxxxxxxxx└─────
+in   : └─────┘     xxxxxxxxxxxx└───────────┘     xxxxxxxxxxxx└─────
 .br
-       ┐     ┌───────────┐
+       ┐     ┌───────────┐
 .br
-on   : └─────┘           └─────────────────────────────────────────
+on   : └─────┘           └─────────────────────────────────────────
 .br
-       ┐                                   ┌───────────┐
+       ┐                                   ┌───────────┐
 .br
-off  : └───────────────────────────────────┘           └───────────
+off  : └───────────────────────────────────┘           └───────────
 .br
-       ┐                 ┌─────────────────────────────┐
+       ┐                 ┌─────────────────────────────┐
 .br
-is-on: └─────────────────┘                             └───────────
+is-on: └─────────────────┘                             └───────────
 
 """;
 
 pin in  bit in;
 pin in  bit is_on;
 pin out bit on;
 pin out bit off;
-variable int old_in;
-variable int to_state=0;
+param rw u32 debounce = 2 "debounce delay in cycles";
```

Comment:
How can we disable debouncing? With debounce = 0 ?
However, isn't it better to have a default value without debouncing to not eventually change the behaviour of existing configurations and add a note like "if you connect a physical button, a debounce value of minimum 2 is strongly recommend!" ?

Reply:
I copied that over from the 'toggle' component which I presumed to be a good example. 
As for existing configs, since the current component is buggy people had to find workarounds (see links above). So if somebody has put an additional 'toggle' component between the button and the 'toggle2nist' component to make it work then the fixed version of it will now require two button presses which is actually what one would expect when we read the description of the component. Not sure how to circumvent this unless we want to keep the broken things and just add the fixed things to it.

Reply:
Adding the debounce is a good think, I agree.
Only thinking that it adds some very little delay but I guess that would hurt nobody.
To be 100% safe we could use the update-ini script to disable the debouncing, but that would be cracking a walnut with a sledgehammer...

Reply:
I get your point and I have no problem with setting the default to 0. The only reason I chose 2 was to make the two 'toggle' components consistent.